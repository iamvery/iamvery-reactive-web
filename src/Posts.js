import React, { Component } from 'react';

class Posts extends Component {
  state = {
    loaded: false,
  };

  componentDidMount() {
    fetch('https://gist.githubusercontent.com/iamvery/a46ba7af5cf5e017c6865bc8455ca05a/raw/60fae242a4d531a767c154f1d5d07560103e6e09/feed.json')
      .then((response) => response.json())
      .then((json) => json.posts)
      .then((posts) => this.setState({posts, loaded: true}))
      .catch(console.log)
  }

  renderPost = (post, index) => <li key={index}><a href={post.link}>{post.title}</a></li>;

  render() {
    if (this.state.loaded) {
      return (
        <ul>
          {this.state.posts.map(this.renderPost)}
        </ul>
      );
    } else {
      return (
        <div>Loading...</div>
      );
    }
  }
}

export default Posts;
