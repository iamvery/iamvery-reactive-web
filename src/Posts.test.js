import React, { Component } from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Posts from './Posts';

it('displays posts', () => {
  const rendered = shallow(<Posts data={['First!', 'Another!']}/>);

  expect(rendered.text()).toContain('First!');
  expect(rendered.text()).toContain('Another!');
});

it('does not change unexpectedly', () => {
  const posts = renderer.create(<Posts data={['one', 'two']} />);
  let tree = posts.toJSON();
  expect(tree).toMatchSnapshot();
});
