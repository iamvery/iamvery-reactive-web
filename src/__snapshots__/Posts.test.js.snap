// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does not change unexpectedly 1`] = `
<ul>
  <li>
    one
  </li>
  <li>
    two
  </li>
</ul>
`;
